app/src/main/java/com/example/returnpals/ScheduleReturn.kt

```diff
@@ -15,69 +25,134 @@ import java.time.LocalDate
- */
-object ScheduleReturn {
+/////////////////////////////////////////////////////////////////////////////
+// PUBLIC API
+////////////////////
 
-    @RequiresApi(Build.VERSION_CODES.O)
-    class ViewModel(init: PickupInfo) : androidx.lifecycle.ViewModel() {
+object ScheduleReturn {}
 
-        private val idManager: IdManager = IdManager()
-        private val _pickup: MutableStateFlow<PickupInfo>
-        val pickup: StateFlow<PickupInfo>
+@Composable
+fun ScheduleReturn(
+    navController: NavController,
+    pickupInfo: PickupInfo,
+) {
 
-        init {
-            _pickup = MutableStateFlow(init)
-            pickup = _pickup.asStateFlow()
-            // _pickup.value and pickup.value should NOT be null after init
-        }
+    val viewModel = remember { ScheduleReturnViewModel(pickupInfo) }
+    val step = remember { mutableIntStateOf(1) }
+    val state = viewModel.pickup.collectAsState();
 
-        fun onChangeDate(value: LocalDate) {
-            _pickup.update {
-                it.date = value
-                it
-            }
-        }
+    val confirm = ConfirmPickup()
+    val thanks = ThankYou()
+
+    val minDate = LocalDate.now().minusDays(1)
+    val maxDate = LocalDate.now().plusMonths(2)
+
+    when (step.intValue) {
+        1 -> PickupDateUI(
+                date = state.value.date,
+                onChangeDate = { state.value.date = it },
+                onClickNext = { step.intValue++ },
+                onClickBack = { step.intValue-- },
+                isValidDate = { state.value.date > minDate && state.value.date < maxDate }
+            )
+        2 -> SelectAddressContent(
+                navController = navController,
+            )
+        3 -> PickupMethodUI(
+                method = state.value.method,
+                onClickNext = {},
+                onClickBack = {},
+                onChangeMethod = {},
+            )
+        4 -> PricingUI(
+                onClickNext = {},
+                onClickBack = {},
+                onChangePlan = {},
+                plan = state.value.pricing,
+            )
+        5 -> PackagesUI(
+                packages = state.value.packages.values.toList(),
+                onAddLabel = {},
+                onRemoveLabel = {},
+                onClickNext = {},
+                onClickBack = {},
+            )
+        6 -> confirm.drawConfirmPickup(
+                typeOfPickup = state.value.method.toString(),
+//            TODO: change type of pickupDate parameter to LocalDate
+//               OR change type of PackageInfo.date to Calendar
+//            TODO: show pricing only if BRONZE plan is selected
+                pickUpAddress = state.value.address!!,
+                nextButton = { step.intValue++ },
+                backButton = { step.intValue-- },
+                promoButton = {}
+            )
+        7 -> thanks.drawThankYouUI(
+                dashBoardButton = { /* TODO: navigate to dashboard */ },
```

Comment:
I think going back to the dashboard describes the end of the State. I think this button should be exposed as a parameter, so the navigation function calls the ReturnSchedule State can set the next navigation after the end of the process. Exposing the end of the process button can help encapsulate the code into a testable piece.  If you keep chaining the navigation for the entire app it could become hard to test as separate components.  This could be a good stop point for this function/State.

Reply:
I agree, I'll do this next pull request I make.